codemod_tox/base.py

```diff
@@ -29,6 +29,27 @@ def startswith(self, prefix: str) -> bool:
         """
         return self.map_all(lambda x: x.startswith(prefix))
 
+    def endswith(self, suffix: str) -> bool:
+        """
+        Returns whether all possibilities end with `suffix`.
+        """
+        return self.map_all(lambda x: x.endswith(suffix))
+
+    def only(self, value: str) -> bool:
+        """
+        Returns whether all possibilities are exactly `value`.
+        """
+        return self.map_all(value.__eq__)
+
+    def one(self) -> str:
+        """
+        Returns the string if only one string matches, otherwise raise ValueError.
+        """
+        s = set(self)
+        if len(s) == 1:
```

Comment:
Could we add a `def __len__()` to the base class and require the children to implement it? That way we're not inserting into a set just to find the length? 

Or maybe I'm misunderstanding the usecase here.

Reply:
This is to detect where a `ToxOption` matches only one string, so it doesn't need to be a `ToxOption` [any more].  For example, `py{38}` doesn't need the curlies, and the set is to catch cases like `py{38,38}` which likewise can be simplified.

This is used below in `_bucket` when finding literal prefix and suffix of a `ToxEnv`.  This was the most obvious way to implement it -- I couldn't come up with a FP way like the `.one` above.  I need both the boolean and the value and doing it the straightforward way requires iterating twice...

Reply:
Maybe

```
if self.map_all(str.__eq__):
  return next(iter(self)
```

Keep in mind this is in `ToxBase` so special-casing `ToxOptions.options[0]` or something would have to be a subclass impl (that doesn't save much, if these only expand to a couple dozen entries).